src/pages/Swap/LimitOrderBox/components/LimitOrderForm/LimitOrderForm.tsx

```diff
@@ -524,6 +517,14 @@ export function LimitOrderForm({ account, provider, chainId }: LimitOrderFormPro
                 )}
                 value={formattedBuyAmount}
                 onUserInput={handleInputOnChange(buyTokenAmount.currency as Token, handleBuyCurrencyAmountChange)}
+                onMax={() => {
+                  if (!buyCurrencyMaxAmount) return
+                  handleBuyCurrencyAmountChange({
+                    currency: buyCurrencyMaxAmount?.currency as Token,
+                    amountWei: buyCurrencyMaxAmount?.raw.toString(),
+                    amountFormatted: buyCurrencyMaxAmount.toSignificant(6),
+                  })
+                }}
```

Comment:
I would declare this function above JSX and just pass the reference here.

Reply:
Forgot about this, fixed now!